src/community/geopkg/src/main/java/org/geoserver/geopkg/wps/GeoPackageProcess.java

```diff
@@ -171,9 +171,8 @@ public URL execute(
 
         String outputName = contents.getName() + ".gpkg";
         if (!remove && path != null) {
-            File urlToFile = URLs.urlToFile(path);
-            urlToFile.mkdirs();
-            file = new File(urlToFile, contents.getName() + ".gpkg");
+            String urlToFile = URLs.urlToFile(path).getPath();
+            file = resources.getExternalOutputFile(urlToFile, outputName);
```

Comment:
Since `resources` is responsible for both getExternalOutputFile and getOutputResource could it take responsibility for this logic to avoid possibility for processes handing this parameter in an inconsistent manner.

`resources` already knows if setExternalOutputDirectory has been provided.

Reply:
The code was moved into WPSResourceManager.